### PR TITLE
Update `maxclients` directive in `redis.conf` info/wording

### PR DESCRIPTION
**Pull request by lukmdo**
Simplify/clarify and align with other config docs

### CODE REVIEW

File: docs/reference/clients.md [hunk 1]
```diff
@@ -47,8 +47,7 @@ However, Redis does the following two things when serving clients:
 In Redis 2.4 there was a hard-coded limit for the maximum number of clients
 that could be handled simultaneously.
 
-In Redis 2.6 and newer, this limit is dynamic: by default it is set to 10000 clients, unless
-otherwise stated by the `maxclients` directive in `redis.conf`.
+In Redis 2.6 and newer, this limit is configurable by `maxclients` directive in `redis.conf` (set by default to 10000 clients).
```
Comment (dwdougherty) [inline docs/reference/clients.md:50]: I hope you don't mind a bit of copy-editing.

```suggestion
In Redis 2.6 and newer, this limit is configurable setting the `maxclients` directive in `redis.conf`. The default is 10,000 clients.
```

Comment (lukmdo) [reply docs/reference/clients.md:50]: Thanks for the suggestion - I like it and applied. 👍

File: docs/reference/clients.md [hunk 2]
```diff
@@ -47,8 +47,7 @@ However, Redis does the following two things when serving clients:
 In Redis 2.4 there was a hard-coded limit for the maximum number of clients
 that could be handled simultaneously.
 
-In Redis 2.6 and newer, this limit is dynamic: by default it is set to 10000 clients, unless
-otherwise stated by the `maxclients` directive in `redis.conf`.
+In Redis 2.6 and newer, this limit is configurable setting the `maxclients` directive in `redis.conf`. The default is 10,000 clients.
```
Comment (dwdougherty) [inline docs/reference/clients.md:50]: Copy-editing my copy-editing. 🤣

```suggestion
In Redis 2.6 and newer, this limit is configurable using the `maxclients` directive in `redis.conf`. The default is 10,000 clients.
```

**Review by dwdougherty — APPROVED** [review]
LGTM.

**Pull request merged**